XamarinCommunityToolkit/Behaviors/RegexValidationBehavior.shared.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Xamarin.Forms;
+using System.Text.RegularExpressions;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    public class RegexValidationBehavior : TextValidationBehavior
+    {
+        public static readonly BindableProperty PatternProperty = BindableProperty.Create(nameof(Pattern), typeof(string), typeof(RegexValidationBehavior), propertyChanged: OnRegexPropertyChanged);
+
+        public static readonly BindableProperty OptionsProperty = BindableProperty.Create(nameof(Options), typeof(RegexOptions), typeof(RegexValidationBehavior), RegexOptions.None, propertyChanged: OnRegexPropertyChanged);
+
+        Regex Regex { get; set; }
+
+        public string Pattern
+        {
+            get => (string)GetValue(PatternProperty);
+            set => SetValue(PatternProperty, value);
+        }
+
+        public RegexOptions Options
```

Comment:
Since RegexOptions is an Enum, that's super convenient to set it through the xaml

```xaml
<local:RegexValidationBehavior Options="ExplicitCapture">
</local:RegexValidationBehavior>
```

Reply:
I thought it was a class of some sorts with all sorts of properties in it. My bad.